### PR TITLE
addpkg(master/faust): A functional programming language for signal pr…

### PR DESCRIPTION
**Pull request by ghost**
Faust has some weird compilation process. The `-DANDROID` is necessary to make sure `execinfo.h` is not included. see `compiler/errors/exception.hh:32`:

```
#if !defined(EMCC) && !defined(WIN32) && !defined(ANDROID) && !defined(ALPINE)
#include <execinfo.h>
#endif
```
I'm wondering if I should patch exception.hh to add `__ANDROID__` instead of defining `-DANDROID`.

### CODE REVIEW

**Comment by TomJo2000** [comment]
> I'm wondering if I should patch exception.hh to add `__ANDROID__` instead of defining `-DANDROID`.

Generally compiler options are preferred over patches when possible since patches tend to break more when a package is being updated, so `-DANDROID` would probably be preferred if both have the same effect.

File: packages/faust/build.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		raqt \
+		linuxunity \
+		jack \
+		jaqt \
+		jackrust \
+		jackconsole; do
+		rm faust2${i}
+	done
+
+	sed -i 's/usage.sh/faustusage.sh' $(find . -type f -exec grep -Iq . {} \+ -print)
```
Comment (TomJo2000) [inline packages/faust/build.sh:76]: You forgot to close this sed substitution, that's what terminated the build process.

**Review by TomJo2000 — CHANGES_REQUESTED** [review]

**Comment by TomJo2000** [comment]
Sorry hit the wrong button, wanted to change my review to an approval.

**Review by TomJo2000 — APPROVED** [review]

**Comment by TomJo2000** [comment]
Hmm build failed with: [`./install: line 3: faust: command not found`](https://github.com/termux/termux-packages/actions/runs/7681508298/job/20934769634?pr=19073#step:8:2723)
I'm guessing `faust` isn't in a directory that is in `$PATH` at this point, maybe you need to move it into `$TERMUX_PREFIX/bin` before calling `./install`.

Maybe something along these lines.
https://github.com/termux/termux-packages/blob/8a82c0f4e4f7cf76b8b9c5bc7708808215d9f4e7/packages/nerdfix/build.sh#L20-L22

**Comment by TomJo2000** [comment]
> won't executing the compiled aarch64/arm in x86 machine during build results in a invalid format? That was the behaviour in the past, I didn't know how things have changed.

¯\\\_(ツ)\_/¯


> (Also closing for now so I can do build on my own CI instead of populating termux actions with faust errors)

Eh we're averaging 20+ CI runs daily, I don't think your PR is cluttering anything up.

**Comment by TomJo2000** [comment]
If you wanna just break out the libraries into separate PR that would also be fine.
The core package seemed to be building just fine.

**Comment by 2096779623** [comment]
> won't executing the compiled aarch64/arm in x86 machine during build results in a invalid format? 

No, because the termux build system has already set up a cross-architecture compilation environment.
```bash
~ $ file faust
faust: ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /system/bin/linker64, stripped
~ $
```

> (Also closing for now so I can do build on my own CI instead of populating termux actions with faust errors)

If the pr is still in WIP, you can convert it to a draft so that the developers don't receive so many emails.

You should replace ./install with `cp *.lib old/*.lib $TERMUX_PREFIX/share/faust/` .

```bash
~ $ faust --dspdir
sh: which: inaccessible or not found
/share/faust
~ $ faust --dspdir
/data/data/com.termux/files/usr/share/faust
```

**Pull request closed without merging**